shell-completion/Makefile

```diff
@@ -1,29 +1,31 @@
+MAKEFLAGS += --jobs=$(shell nproc)
 
 .PHONY: command-options clean
 
 all: hledger-completion.bash
 
-hledger-completion.bash: hledger-completion.bash.m4 commands-list.txt query-filters.txt generic-options.txt command-options
+hledger-completion.bash: hledger-completion.bash.m4 query-filters.txt generic-options.txt command-options
 	m4 hledger-completion.bash.m4 > $@
 
 generic-options.txt:
 	hledger -h | ./output-options.sh | sort -u > $@
 
-commands.txt:
-	hledger | ./output-commands.sh | grep -v ^hledger | sort -u > $@
-	echo ui  >> $@
-	echo web >> $@
-	echo api >> $@
+COMMANDS := $(sort $(shell hledger | ./output-commands.sh | grep -v ^hledger | sort -u) ui web api)
 
-commands-list.txt: commands.txt
-	paste -sd, $^ | tr -d '\n' > $@
+space := $() $()
+comma := ,
+
+commands-list.txt:
+	echo -n $(patsubst $(space),$(comma),$(COMMANDS)) > $@
```

Comment:
I don't think this would work because `patsubst` acts on space separated words, not the spaces themselves

Reply:
And I wouldn't worry about `paste`. I think it's pretty much a given on any system.

Reply:
Fair enough. There is a way to force that, but shelling out to something else is undoubtedly less hackish looking. I'll use `sed` though (required elsewhere anyway) instead of `paste`.

Reply:
This should work too:
```make
commands-list.txt:
	printf %s $(subst $(space),$(comma),$(COMMANDS)) > $@
```
As a rule of thumb, whenever one needs to pass an option to `echo`, `printf` is the more portable choice.